src/Codec/Xlsx/Formatted.hs

```diff
@@ -227,14 +229,31 @@ data Formatted = Formatted {
 -- then 'minimalStyleSheet' is a good choice.
 formatted :: Map (Int, Int) FormattedCell -> StyleSheet -> Formatted
 formatted cs styleSheet =
-   let initSt         = stateFromStyleSheet styleSheet
-       (cs', finalSt) = runState (mapM (uncurry formatCell) (M.toList cs)) initSt
-       styleSheet'    = updateStyleSheetFromState styleSheet finalSt
-   in Formatted {
-          formattedCellMap    = M.fromList (concat cs')
-        , formattedStyleSheet = styleSheet'
-        , formattedMerges     = reverse (finalSt ^. formattingMerges)
-        }
+  let ([ws], styleSheet') = formatWorksheets (cs:[]) styleSheet
+  in Formatted {
+       formattedCellMap    = _wsCells ws
+     , formattedStyleSheet = styleSheet'  
+     , formattedMerges     = _wsMerges ws
+     }
+
+formatWorksheets :: [Map (Int, Int) FormattedCell] -> StyleSheet -> ([Worksheet], StyleSheet)
+formatWorksheets fcss initStyle = foldr formatSingle ([], initStyle) fcss
```

Comment:
I bet `xlslx` needs more care about strictness in other places but I think we'd better use `foldl'` here. It will also fill styles from left to right and in reverse order as `foldr` will do.
Otherwise I'm OK with this change and will release it as `0.7.2` with plans to deprecate `formatted` in `0.8` and remove in `0.9`